File: app/src/main/java/com/ianfield/bodyscoring/activity/SettingsActivity.java
package com.ianfield.bodyscoring.activity;

import android.os.Bundle;
import android.preference.Preference;
import android.preference.PreferenceFragment;
import android.support.v7.app.AppCompatActivity;

import com.ianfield.bodyscoring.R;

/**
 * Created by Ian on 09/01/2016.
 */
public class SettingsActivity extends AppCompatActivity {

    SettingsFragment settingsFragment;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_settings);

        settingsFragment = new SettingsFragment();
        getFragmentManager().beginTransaction()
                .replace(R.id.container, settingsFragment)
                .commit();
    }

    public static class SettingsFragment extends PreferenceFragment {
        @Override
        public void onCreate(Bundle savedInstanceState) {
            super.onCreate(savedInstanceState);

            addPreferencesFromResource(R.xml.settings);

            Preference locality = findPreference(getString(R.string.pref_locality));

            String value = locality.getSharedPreferences().getString(getString(R.string.pref_locality), getString(R.string.pref_localities_default));
            locality.setSummary(value);

            locality.setOnPreferenceChangeListener((preference, newValue) -> {
                if (preference.getKey().equals(getString(R.string.pref_locality))) {
                    preference.setSummary((String) newValue);
                }
                return true;
            });

        }
    }


}

File: app/src/main/java/com/ianfield/bodyscoring/activity/SettingsActivity.kt
package com.ianfield.bodyscoring.activity

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.preference.Preference
import androidx.preference.PreferenceFragmentCompat
import com.ianfield.bodyscoring.R

/**
 * Created by Ian on 09/01/2016.
 */
class SettingsActivity : AppCompatActivity() {

    lateinit var settingsFragment: SettingsFragment

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_settings)

        settingsFragment = SettingsFragment()
        supportFragmentManager.beginTransaction()
                .replace(R.id.container, settingsFragment)
                .commit()
    }

    class SettingsFragment : PreferenceFragmentCompat() {
        override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
            addPreferencesFromResource(R.xml.settings)

            val locality: Preference? = findPreference(getString(R.string.pref_locality))
            val value = locality?.sharedPreferences?.getString(getString(R.string.pref_locality), getString(R.string.pref_localities_default))
            locality?.summary = value
            locality?.setOnPreferenceChangeListener { preference, newValue ->
                if (preference.key == getString(R.string.pref_locality)) {
                    preference.summary = newValue as String
                }
                true
            }

        }
    }


}
